src/services/source-statuses.service.ts

```diff
@@ -1,7 +1,67 @@
+import { NotUniqueError } from '../errors/not-unique.error';
+import { NotFoundError } from '../errors/not-found.error';
+import { RemovalRestrictedError } from '../errors/removal-restricted.error';
 import { SourceStatus } from '../models/source-status.model';
 
 export class SourceStatusesService {
+  /** GET all source statuses */
   static async getAll() {
     return await SourceStatus.find();
   }
+
+  /** GET one source status */
+  static async getOneById(source_status_id: number): Promise<SourceStatus> {
+    const source_status = await SourceStatus.findOne({ id: source_status_id });
+
+    if (!source_status) throw new NotFoundError('Source status not found');
+    return source_status;
+  }
+
+  /** POST new source status */
+  static async addOne(name: string, description: string): Promise<SourceStatus> {
+    const new_source_status = new SourceStatus();
+    new_source_status.name = name;
+    new_source_status.description = description;
+
+    return this.saveSourceStatus(new_source_status);
+  }
+
+  /** PUT updated source status */
+  static async updateOne(source_status_id: number, name: string, description: string): Promise<SourceStatus> {
+    const updated_source_status = await SourceStatus.findOne({ id: source_status_id });
+
+    if (!updated_source_status) throw new NotFoundError('Source status not found');
+
+    if (name !== undefined) updated_source_status.name = name;
+    if (description !== undefined) updated_source_status.description = description;
+
+    return this.saveSourceStatus(updated_source_status);
+  }
+
+  /** DELETE source status */
+  static async deleteOne(source_status_id: number) {
+    const deleted_source_status = await SourceStatus.findOne({ id: source_status_id });
+    if (!deleted_source_status) throw new NotFoundError('Source status not found');
+    try {
+      await deleted_source_status.remove();
+    } catch (error) {
+      if (error.message.includes('FOREIGN KEY constraint failed')) {
+        throw new RemovalRestrictedError('The source status can not be deleted, it is used by at least one source');
```

Comment:
Error message here could contain details about the source status (e.g. its name and id).

Reply:
Only id property is provided in deleteOne method, so I have added only id to the message.